src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -198,7 +198,9 @@ impl<'a> JobQueue<'a> {
         }
 
         let build_type = if self.is_release { "release" } else { "debug" };
-        let profile = cx.lib_profile(&cx.current_package);
+        let profile = cx.ws.current_opt().map_or_else(Profile::default, |p| {
+            cx.lib_profile(p.package_id()).to_owned()
+        });
```

Comment:
The package argument here to `lib_profile` is actually ignored (long story), so let's just go ahead and remove that argument and then we can avoid the workaround here.